test/common/cdp.py

```diff
@@ -226,3 +228,25 @@ def set_frame(self, frame):
         self.cur_frame = frame
         if self.trace:
             print("-> switch to frame %s" % frame)
+
+    def get_js_log(self):
+        """Return the current javascript console log"""
+
+        if self.valid:
+            # needs to be wrapped in Promise
+            messages = self.command("new Promise((resolve, reject) => resolve(messages))")
+            return map(lambda m: "%s: %s" % tuple(m), messages)
+        return []
+
+    def read_log(self):
+        """Returns an iterator that produces log messages one by one. Blocks if
+        there are no new messages right now."""
```

Comment:
Short description of a docstring needs to be a single line. I. e. put a paragraph between the sentences.

Reply:
+1